core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -215,8 +232,32 @@ public static String encodeNamespace(Namespace ns) {
    * @return a namespace
    */
   public static Namespace decodeNamespace(String encodedNs) {
```

Comment:
This function looks dangerous now.

Reply:
can you please elaborate what you mean by "dangerous" here? Nothing really changed for callers of this method

Reply:
Mixing this and the new function.

Reply:
it's still not clear to my why that would be "dangerous". There is no behavioral change for existing callers of this method. Can you please add a concrete example/explanation to justify the "dangerous" part?

Reply:
Users can mix both - causing "confusion".

Reply:
I don't think that argument is actually true. **Users** won't be using this code as this code is used on the client and the server but not by **users** in the classical sense of a user that uses Iceberg. For engine/catalog implementers the javadoc states that one should have used `encodeNamespace(Namespace namespace)` if you use `decodeNamespace(String encodedNs)`.
Using words like `dangerous` and `confusing` without clear examples and justifications isn't helpful in a code review